feature/navigation/drawer/dropdown/src/main/kotlin/net/thunderbird/feature/navigation/drawer/dropdown/domain/usecase/GetDisplayTreeFolder.kt

```diff
@@ -0,0 +1,103 @@
+package net.thunderbird.feature.navigation.drawer.dropdown.domain.usecase
+
+import app.k9mail.core.mail.folder.api.Folder
+import app.k9mail.core.mail.folder.api.FolderType
+import kotlinx.collections.immutable.persistentListOf
+import kotlinx.collections.immutable.toImmutableList
+import net.thunderbird.feature.navigation.drawer.dropdown.domain.DomainContract.UseCase
+import net.thunderbird.feature.navigation.drawer.dropdown.domain.entity.DisplayAccountFolder
+import net.thunderbird.feature.navigation.drawer.dropdown.domain.entity.DisplayFolder
+import net.thunderbird.feature.navigation.drawer.dropdown.domain.entity.DisplayTreeFolder
+import net.thunderbird.feature.navigation.drawer.dropdown.domain.entity.DisplayUnifiedFolder
+
+internal class GetDisplayTreeFolder : UseCase.GetDisplayTreeFolder {
+
+    override fun invoke(folders: List<DisplayFolder>, maxDepth: Int): DisplayTreeFolder {
+        val unifiedFolderTreeList = folders.filterIsInstance<DisplayUnifiedFolder>().map {
+            DisplayTreeFolder(
+                displayFolder = it,
+                displayName = it.unifiedType.id,
+                totalUnreadCount = it.unreadMessageCount,
+                totalStarredCount = it.starredMessageCount,
+                children = persistentListOf(),
+            )
+        }
+
+        val accountFolders = folders.filterIsInstance<DisplayAccountFolder>().map {
+            val path = flattenPath(it.folder.name, maxDepth)
+            println("Flattened path for ${it.folder.name} → $path")
+            path to it
+        }
+        val accountFolderTreeList = buildAccountFolderTree(accountFolders)
+
+        return DisplayTreeFolder(
+            displayFolder = null,
+            displayName = null,
+            totalUnreadCount = accountFolderTreeList.sumOf { it.totalUnreadCount },
+            totalStarredCount = accountFolderTreeList.sumOf { it.totalStarredCount },
+            children = (unifiedFolderTreeList + accountFolderTreeList).toImmutableList(),
+        )
+    }
+
+    private fun flattenPath(folderName: String, maxDepth: Int): List<String> {
+        val parts = folderName.split("/").map { it.takeIf { it.isNotBlank() } ?: "(Unnamed)" }
```

Comment:
I know I am late because this is already merged (thank you so much btw), but could the separator be made configurable? I am not familiar with the IMAP spec, but at least my mail provider seems to use `.` as separator for foldernames instead of `/`. A workaround/fix for this could be to "normalize" the folder names by simply replacing `.` with `/` before the tree is generated and keep using `/` throughout the code.

side note: using a const for the separator sign might be a good idea.

Reply:
Please file a new issue on this, we'd have to create a new PR anyway :)